r_scripts/R_functions.R

```diff
@@ -157,3 +157,91 @@ filter_times <- function(times, df){
     return(out)
 }
 #________________________________________________________
+
+#________________________________________________________
+# plot mass based emission factors 
+plot_mass_ef <- function(df, pol_name, type){
+  
+  p1 <- ggplot(df, aes(x = stove_fuel, y = mass_ef, colour = fuel)) +
+    geom_point(size = 1) +
+    facet_grid(pol ~ stovecat, scales = 'free') +
+    ggtitle(paste(pol_name, "ef by stove/fuel combination")) +
+    xlab("stove type") +
+    ylab("mass based emissions factors (mg/kg of fuel)") +
+    theme_minimal() +
+    scale_x_discrete(label=function(x) sub(" [: : :].*", "", x)) +
+    theme(text = element_text(size=18),
+          axis.text.x = element_text(angle = 45, vjust = 1.18, hjust = 1, size=7.5),
+          panel.spacing = unit(2, "lines"))
+  
+  p2 <- ggplot(df, aes(x = stove, y = mass_ef, colour = fuel)) +
+    geom_point(size = 1) +
+    facet_grid(pol ~ stovecat, scales = 'free') +
+    ggtitle(paste(pol_name, "ef by stove type")) +
+    xlab("stove type") +
+    ylab("mass based emissions factors (mg/kg of fuel)") +
+    theme_minimal() +
+    theme(text = element_text(size=18),
+          axis.text.x = element_text(angle = 45, vjust = 1.1, hjust = 1, size=10),
+          panel.spacing = unit(2, "lines"))
+  
+  p3 <- ggplot(df, aes(x = fuel, y = mass_ef, colour = stove)) +
+    geom_point(size = 1) +
+    facet_grid(pol ~ fuelcat, scales = 'free') +
+    ggtitle(paste(pol_name, "ef by fuel type")) +
+    xlab("fuel type") +
+    ylab("mass based emissions factors (mg/kg of fuel)") +
+    theme_minimal() +
+    theme(text = element_text(size=18),
+          axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1, size=14),
+          panel.spacing = unit(2, "lines"))
+  
+  print(p1)
```

Comment:
you can use library(gridExtra) -> grid.arrange() for more control of your plot layouts